hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/redis/AssertionExpirationCacheRedisCodec.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.ida.hub.samlengine.redis;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.util.ByteBufferBackedInputStream;
+import com.google.common.io.ByteStreams;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+public class AssertionExpirationCacheRedisCodec extends ExpirationCacheRedisCodec<String> {
+
+    public AssertionExpirationCacheRedisCodec(ObjectMapper objectMapper) {
+        super(objectMapper);
+    }
+
+    @Override
+    public String decodeKey(ByteBuffer keyBytes) {
+        try {
+            InputStream inputStream = new ByteBufferBackedInputStream(keyBytes);
+            // TODO: Replace with `inputStream.readAllBytes()` when we only use Java 9+
+            return new String(ByteStreams.toByteArray(inputStream));
```

Comment:
```suggestion
            return new String(ByteStreams.toByteArray(inputStream), StandardCharsets.UTF_8);
```
oops, missed this one